src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs

```diff
@@ -1692,14 +1692,29 @@ internal sealed class ListFacetsChecker : FacetsChecker
 
         internal override bool MatchEnumeration(object value, ArrayList enumeration, XmlSchemaDatatype datatype)
         {
-            for (int i = 0; i < enumeration.Count; ++i)
+            Array values = (value as Array)!;
+            Datatype_List list = (datatype as Datatype_List)!;
+            Debug.Assert(list != null);
+
+            for (int j = 0; j < values.Length; ++j)
             {
-                if (datatype.Compare(value, enumeration[i]!) == 0)
+                bool found = false;
+                for (int i = 0; i < enumeration.Count; ++i)
                 {
-                    return true;
+                    Array enumValue = (enumeration[i] as Array)!;
```

Comment:
it's weird this is an array, possibly a follow up PR to fix that would make sense